lib/server/services/run/RunService.js

```diff
@@ -276,13 +276,39 @@ class RunService {
  * @returns {Promise<undefined|Error>} - promise on result of db queries
  */
 const updateEorReasonsOnRun = async (runId, userName, eorReasonPatches) => {
-    let reasonTypes = await ReasonTypeRepository.findAll();
-    reasonTypes = reasonTypes.map((reason) => reason.id);
+    const reasonTypes = await ReasonTypeRepository.findAll();
 
-    const allReceivedReasonsExists = eorReasonPatches.every((eorReasonPatch) => reasonTypes.includes(eorReasonPatch.reasonTypeId));
+    // Determine the format of the eorReasonPatches (can be reasonTypeId || category and title)
+    if (eorReasonPatches[0]?.reasonTypeId) {
+        const reasonTypeIds = reasonTypes.map((reason) => reason.id);
+        const allReceivedReasonsExists = eorReasonPatches.every((eorReasonPatch) => reasonTypeIds.includes(eorReasonPatch.reasonTypeId));
 
-    if (!allReceivedReasonsExists) {
-        throw new Error('Provided reason types do not exist');
+        if (!allReceivedReasonsExists) {
+            throw new Error('Provided reason types do not exist');
```

Comment:
Currently you are throwing an error if the EOR is not as expected.
While this is correct for the scenario in which the user selects manually the EOR, and we want to throw an error if the selection is bad, it is not the same for automated EOR set via the AliECS Core where we want to store the critical information such as `o2_time_stop`

To differentiate the 2 scenarios, you should split the grPCController method to make 2 calls to the runservice.
- one that updates the run with received information but EOR reasons. 
- one that updates the run with the recieved EOR reasons only. (this now can throw as we are not interested if it fails or not).
But the first scenario needs to always save the information, that is why we should remove EOR from the initial call.

Reply:
Should be fixed now